testsuites/syncgateway/functional/tests/test_log_redaction.py

```diff
@@ -0,0 +1,454 @@
+    3. Create user in sync_gateway
+    4. Create docs with xattrs
+    5. Do a post call for sgCollect With Rest Api and verify rest API works
+    """
+
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    cluster_helper = ClusterKeywords(cluster_config)
+    cluster_hosts = cluster_helper.get_cluster_topology(cluster_config)
+    sg_admin_url = cluster_hosts["sync_gateways"][0]["admin"]
+    sg_url = cluster_hosts["sync_gateways"][0]["public"]
+    sg_ip = host_for_url(sg_admin_url)
+    sg_platform = params_from_base_test_setup["sg_platform"]
+    sg_db = "db"
+    num_of_docs = 10
+    user_name = 'autotest'
+    password = 'validkey'
+    if get_sync_gateway_version(sg_ip)[0] < "2.1":
+        pytest.skip("log redaction feature not available for version < 2.1 ")
+
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    # Modifying log redaction level to partial
+    temp_cluster_config = copy_to_temp_conf(cluster_config, mode)
+    persist_cluster_config_environment_prop(temp_cluster_config, 'redactlevel', "partial", property_name_check=False)
+    cluster = Cluster(config=temp_cluster_config)
+    cluster.reset(sg_config_path=sg_conf)
+
+    # Get sync_gateway host
+    cluster = load_cluster_config_json(cluster_config)
+    sg_host = cluster["sync_gateways"][0]["ip"]
+
+    # 3. Create user in sync_gateway
+    sg_client = MobileRestClient()
+    channels = ["log-redaction"]
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name=user_name, password=password, channels=channels)
+    autouser_session = sg_client.create_session(url=sg_admin_url, db=sg_db, name=user_name, password=password)
+
+    # 4. Create docs with xattrs
+    sgdoc_bodies = document.create_docs(doc_id_prefix='sg_docs', number=num_of_docs,
+                                        attachments_generator=attachment.generate_2_png_10_10, channels=channels)
+    sg_client.add_bulk_docs(url=sg_url, db=sg_db, docs=sgdoc_bodies, auth=autouser_session)
+    assert len(sgdoc_bodies) == num_of_docs
+    if output_dir:
+        if sg_platform == "windows":
+            directory = "c{}\\test".format(":")
+        else:
+            directory = "/home/sync_gateway/data"
+    else:
+        directory = None
+    if redaction_salt:
+        salt_value = "customized-redaction-salt-value"
+        resp = sg_client.sgCollect_info(sg_host, redact_level=redaction_level, redact_salt=salt_value, output_directory=directory)
+    else:
+        resp = sg_client.sgCollect_info(sg_host, redact_level=redaction_level, output_directory=directory)
+    if resp["status"] != "started":
+        assert False, "sg collect did not started"
+
+    count = 0
+    log_info("sg collect is running ........")
+    while sg_client.get_sgCollect_status(sg_host) == "running" and count < 50:
+        time.sleep(1)
+        count += 1
+    time.sleep(5)  # sleep until zip files created with sg collect rest end point
```

Comment:
Is there a better way? This might not be enough on slow machines

Reply:
No, I tried waiting until sgcollect restAPI status says stopped, but still it is taking some time to create zip files after status says stopped